jsonschema-hyperschema.xml

```diff
@@ -653,6 +662,91 @@
+                    normally begin.
+                </t>
+                <t>
+                    For each property name in the object that matches a variable name in the
+                    "href" URI Template, the value of that property adjusts the starting position
+                    of variable resolution for that variable.  Properties which do not match
+                    "href" template variable names MUST be ignored.
+                </t>
+                <figure>
+                    <preamble>
+                        Consider the following schema for an n-ary tree using the path expansion
+                        URI Template syntax:
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+    "type": "object",
+    "required": ["id"],
+    "properties": {
+        "id": {"type": "integer", "minimum": 0},
+        "children": {"type": "array", "items": {"$ref": "#"}}
+    },
+    "links": [
+        {
+            "rel": "self",
+            "href": "/trees{/root,parent,id}"
+            "hrefPointers": {
+                "root": "/id",
+                "parent": "2/id"
+            }
+        }
+    ]
+}]]>
+                    </artwork>
+                    <postamble>
+                        The self links here should always include the root node id,
+                        followed by an immediate parent id if one exists, and always ending
+                        with the id of the node that is the context of the self link.
+                        The context node's id resolves with the standard process and
+                        does not appear in "hrefPointers".  The root id is located
+                        using an absolute JSON Pointer, while the parent node
+                        uses a Relative JSON Pointer, going up two levels (one level up
+                        is the array of children, two levels up is the whole parent node).
+                    </postamble>
+                </figure>
+                <figure>
+                    <preamble>
+                        Given the following instance:
+                    </preamble>
+                    <artwork>
+<![CDATA[{
+  "id": 0,
+  "children": [ {
+    "id": 1,
+    "children": [ {
+      "id": 2,
+      "children": [ {
+        "id": 3
+} ] } ] } ] }]]>
+                    </artwork>
+                </figure>
+                <t>
+                    We get the following self links for each node in the instance as identifed
+                    by JSON Pointers:
+                    <list style="hanging">
+                        <t hangText='"" (the root node)'>/trees/0/0</t>
+                        <t hangText='"/children/0/children/0"'>/trees/0/0/1</t>
+                        <t hangText='"/children/0/children/0/children/0"'>/trees/0/1/2</t>
+                        <t hangText='"/children/0/children/0/children/0/children/0"'>/trees/0/2/3</t>
```

Comment:
I'm trying to apply the explanation in `<postamble>` literally and can't get the last three examples working:
* "/children/0/children/0" points to `{"id": 2, "children": [{"id": 3}]}`, so it's URI should be `/trees/0/1/2` because it's parent has "id=1" and the node has "id=2"
* "/children/0/children/0/children/0" points to `{"id": 3}`, so it's URI should be `/trees/0/2/3` (parent's id=2, context node's id=3)
* "/children/0/children/0/children/0/children/0" does not resolve

Reply:
OMG I am so sorry, all of these have an extra /children/0 in them, no wonder none of this makes sense to you.  :-(

I'll update this ASAP, but you can see that your results work perfectly if you chop off that extra path component.  I think this was a copy-paste error from when I had more levels in the tree, and for some reason I cut the wrong part of the example when trimming it down.

Reply:
> Similarly, I'd expect /trees/0/1/2/3 to be the URI of inner node ({"id": 3}), but the href model does not allow this.

This is where I always have trouble with examples.  Perhaps this is just because the current example literally makes no sense the way I messed it up.  But I find that so many people just want to redesign the example to do something else instead of reading the premise.

The intention was specifically *not* to do a normal tree layout.

The point is to show the need for both absolute and relative pointers.  I'm not attached to this example, it just (when written properly) shows it more concisely than anything else.  I try not to do this one with collections because inevitably, someone complains "I just wouldn't do collections that way."

I hate writing examples, and everyone always hates my examples :-P

Reply:
Although you bring up an interesting point that it's not possible to do that "normal' tree layout with `hrefPointers` AFAIK.  Hmm... I don't think we absolutely need to solve every URI construction approach, but it's worth thinking about.